flow/pkg/mongo/validation.go

```diff
@@ -107,6 +107,31 @@ func ValidateOplogRetention(ctx context.Context, client *mongo.Client) error {
 	}
 }
 
+func ValidateCollections(ctx context.Context, client *mongo.Client, sourceTableIdentifiers []string) error {
+	databaseCollectionsMapping := make(map[string][]string)
+
+	for _, sourceTableIdentifier := range sourceTableIdentifiers {
+		database, collection, hasDot := strings.Cut(sourceTableIdentifier, ".")
```

Comment:
Can move `SchameTable`  and `ParseSchemaTable` from `flow/connectors/utils/identifiers.go` to `pkg` and have `ValidateCollections` take `[]SchemaTable` because in the other codebase we have them as separate strings. `ParseSchemaTable` wouldn't be needed here exactly but would be good to align with #5131 over there (`utils.go`)